be/src/vec/functions/function_from_unixtime.cpp

```diff
@@ -0,0 +1,301 @@
+            if constexpr (argument_num == ArgumentNumber::Two) {
+                return {std::make_shared<vectorized::DataTypeInt64>(),
+                        std::make_shared<vectorized::DataTypeString>()};
+            } else {
+                return {std::make_shared<vectorized::DataTypeInt64>()};
+            }
+        } else if constexpr (std::is_same_v<from_type, Float64>) {
+            if constexpr (argument_num == ArgumentNumber::Two) {
+                return {std::make_shared<vectorized::DataTypeFloat64>(),
+                        std::make_shared<vectorized::DataTypeString>()};
+            } else {
+                return {std::make_shared<vectorized::DataTypeFloat64>()};
+            }
+        }
+    }
+
+    DataTypePtr get_return_type_impl(const ColumnsWithTypeAndName& arguments) const override {
+        if (arguments.empty()) {
+            throw doris::Exception(ErrorCode::INVALID_ARGUMENT,
+                                   "from_unixtime() requires at least 1 argument, 0 provided.");
+        }
+
+        if constexpr (argument_num == ArgumentNumber::Two) {
+            if (arguments.size() != 2) {
+                throw doris::Exception(ErrorCode::INTERNAL_ERROR,
+                                       "from_unixtime() requires 2 arguments, {} provided.",
+                                       arguments.size());
+            }
+        }
+
+        const TypeIndex type_idx = arguments[0].type->get_type_id();
+
+        if constexpr (IsDecimalNumber<from_type>) {
+            if (type_idx != TypeIndex::Decimal32 && type_idx != TypeIndex::Decimal64 &&
+                type_idx != TypeIndex::Decimal128V3) {
+                throw doris::Exception(ErrorCode::INTERNAL_ERROR,
+                                       "could not use {} as the first argument of from_unixtime()",
+                                       arguments[0].type->get_name());
+            }
+        } else if constexpr (std::is_same_v<from_type, Int64>) {
+            if (type_idx != TypeIndex::Int64) {
+                throw doris::Exception(ErrorCode::INTERNAL_ERROR,
+                                       "could not use {} as the first argument of from_unixtime()",
+                                       arguments[0].type->get_name());
+            }
+        } else {
+            throw doris::Exception(ErrorCode::INTERNAL_ERROR, "Illegal column type");
+        }
+
+        return make_nullable(std::make_shared<DataTypeString>());
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) const override {
+        const ColumnPtr source_col = block.get_by_position(arguments[0]).column;
+        const DataTypePtr data_type = block.get_by_position(arguments[0]).type;
+        const cctz::time_zone& time_zone = context->state()->timezone_obj();
+
+        const auto* nullable_column = check_and_get_column<ColumnNullable>(source_col.get());
```

Comment:
if so, maybe we should return override `default_impl_for_null` to false.
or you can just ignore the business of null input. just return `ColumnNullable` which nullmap gets true from the function logic, like out-of-bound or something. then at last our default impl framework will combine the nullmap with the input nullmap.

Reply:
makes sence